### ISSUE TITLE
Sketch does not have keyboard focus at start-up

### ISSUE THREAD

**Issue by processing-bugs** [processing/processing#318]
_Original author: m...@christian-thiemann.de (June 27, 2010 00:58:55)_

I am using MacOSX 10.6.4 and this is apparently related to issue 258 (the Java update).

Since Processing 0185 a sketch does not capture keyboard input initially.  Mouse tracking (mouseX/mouseY) works, but keyboard events are only processed after clicking once into the drawing area.

In PApplet.java at line 1512 in the current svn, &quot;requestFocus();&quot; is commented out with reference to issue 258.  Reactivating and changing this line to &quot;requestFocusInWindow();&quot; solves the problem for me (and no PDE focus issues – see below).

Out of curiosity, maybe somebody can enlighten me: Initially I tried adding &quot;requestFocus();&quot; as the first line in draw() in my sketch and that worked well.  So I thought I'd quickly add this to PApplet somewhere and saw the commented-out line.  Although that's probably for a good reason, I tried to reactivate the line.  Strangely, then the focus of the PDE Editor(!) was messed up (i.e., the bug about the Find window not automatically getting focus when called).  So, I tried to move the requestFocus() line into PApplet.handleDraw() right before the call to draw() – thus it's virtually at the same position as if I put it at the beginning of my sketch's draw().  Still the same PDE focus issues. How can that be? Is the PDE a PApplet instance? I am totally puzzled. Thanks for any explanation on this.

_Original issue: http://code.google.com/p/processing/issues/detail?id=279_
